### PR TITLE
Add ARP entries for RubyInstallerTeam.Ruby.3.1 version 3.1.0-1

### PR DESCRIPTION
**Pull request by russellbanks**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/66863)

### CODE REVIEW

File: manifests/r/RubyInstallerTeam/Ruby/3/1/3.1.0-1/RubyInstallerTeam.Ruby.3.1.installer.yaml [hunk 1]
```diff
@@ -18,8 +18,14 @@ Installers:
 - Architecture: x64
   InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-3.1.0-1/rubyinstaller-3.1.0-1-x64.exe
   InstallerSha256: D67F7ECC6C5D82CFB96DB2F3C47CDBB58F2EBEDF8BB1F7D9B047F3D21F39350A
+  AppsAndFeaturesEntries:
+    DisplayName: Ruby 3.1.0-1-x64-ucrt
+    ProductCode: RubyInstaller-3.1-x64-mingw-ucrt_is1
```
Comment (OfficialEsco) [inline manifests/r/RubyInstallerTeam/Ruby/3/1/3.1.0-1/RubyInstallerTeam.Ruby.3.1.installer.yaml:23]:
```suggestion
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.1.0-1-x64-ucrt
    ProductCode: RubyInstaller-3.1-x64-mingw-ucrt_is1
```

File: manifests/r/RubyInstallerTeam/Ruby/3/1/3.1.0-1/RubyInstallerTeam.Ruby.3.1.installer.yaml [hunk 2]
```diff
@@ -18,8 +18,14 @@ Installers:
...
 - Architecture: x86
   InstallerUrl: https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-3.1.0-1/rubyinstaller-3.1.0-1-x86.exe
   InstallerSha256: 409C764E3B3CD6DBA3976B1AAC5C21BF40C5005F2574CBCDCE09775BCDE9A04E
+  AppsAndFeaturesEntries:
+    DisplayName: Ruby 3.1.0-1-x86-msvcrt
+    ProductCode: RubyInstaller-3.1-i386-mingw32_is1
```
Comment (OfficialEsco) [inline manifests/r/RubyInstallerTeam/Ruby/3/1/3.1.0-1/RubyInstallerTeam.Ruby.3.1.installer.yaml:29]:
```suggestion
  AppsAndFeaturesEntries:
  - DisplayName: Ruby 3.1.0-1-x86-msvcrt
    ProductCode: RubyInstaller-3.1-i386-mingw32_is1
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]
Hopefully this was a manual test 👀

**Comment by Trenly** [comment]
cd <path to Ruby>
gci -recurse -file -filter *.installer.yaml | % { Add-ARPEntries.ps1 $_.PSParentPath; $b = New-Guid; git switch upstream/master; git add .; git commit -m “Add ARP Entries for Ruby”; git switch -c $b; git push --set-upstream origin $b; gh pr create -f;}
```

**Comment by russellbanks** [comment]
> ```
> cd <path to Ruby>
> gci -recurse -file -filter *.installer.yaml | % { Add-ARPEntries.ps1 $_.PSParentPath; $b = New-Guid; git switch upstream/master; git add .; git commit -m “Add ARP Entries for Ruby”; git switch -c $b; git push --set-upstream origin $b; gh pr create -f;}
> ```

Wow, thank you - only issue with running it like this is that the Add ARP script doesn't generate the ideal file (it would fail if I didn't clean it up before committing)

**Review by Trenly — APPROVED** [review]

**Pull request merged**